### PR TITLE
Send solved/closed notification when solving/closing ticket from followup

### PR DESCRIPTION
**Pull request by nsalguero**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: inc/itilfollowup.class.php [hunk 1]
```diff
@@ -330,18 +336,16 @@ function post_addItem() {
          $update['id']     = $parentitem->fields['id'];
 
          // don't notify on ITILObject - update event
-         $update['_disablenotif'] = true;
+         if (!$donotif
+             || ($this->input['_status'] != CommonITILObject::SOLVED
+                 && $this->input['_status'] != CommonITILObject::CLOSED)) {
```
Comment (cedric-anne) [inline inc/itilfollowup.class.php:341]: Changes and Problems have multiple statuses that matches the `solved` state.
```suggestion
             || (!in_array($this->input['_status'], $this->getSolvedStatusArray())
                 && !in_array($this->input['_status'], $this->getClosedStatusArray()))) {
```

File: inc/itilfollowup.class.php [hunk 2]
```diff
@@ -254,6 +254,12 @@ function post_addItem() {
          ]);
       }
 
+      if ($donotif) {
```
Comment (cedric-anne) [inline inc/itilfollowup.class.php:257]: Maybe it should be placed after the next block. Indeed, `$donotif = false; // Done for ITILObject update (new status)` have to be evaluated first.

**Comment by cedric-anne** [comment]
Technically, it is OK, but, due to recent changes on master, it is no longer possible to change status of an issue to solved/closed from a followup. I do not know is we will reintroduce this feature.
Ping @orthagh .

**Comment by cconard96** [comment]
Any chance this could be configurable? I use the feature to close/solve tickets at the same time I add followups and tasks a lot. I'm not a fan of getting double emails when both notifications use the same template and contain the new info, and I know the other people on the ticket don't like it either.

**Comment by arthurrschaefer** [comment]
> Any chance this could be configurable? I use the feature to close/solve tickets at the same time I add followups and tasks a lot. I'm not a fan of getting double emails when both notifications use the same template and contain the new info, and I know the other people on the ticket don't like it either.

Indeed, that's a good suggestion.
There are clients who like this closing when adding a follow-up, task, and some want to deliberately block it.

**Comment by cedric-anne** [comment]
It is not anymore possible to change the status while adding a follow-up or a task, so I close this PR.

**Pull request closed without merging**